js/capture.js

```diff
@@ -19,13 +27,61 @@ const capture = {
       });
   },
 
+  // In the parsed whitelist JSON object (from disconnect-entitylist.json):
+  //   - 'properties' keys are first parties
+  //   - 'resources' keys are third parties
+  reformatList(whiteList) {
+    const firstPartyWhiteList = {};
+    const thirdPartyWhiteList = {};
+    let counter = 0;
+    for (const siteOwner in whiteList) {
+      const firstParties = whiteList[siteOwner].properties;
+      for (let i = 0; i < firstParties.length; i++) {
+        firstPartyWhiteList[firstParties[i]] = counter;
+      }
+      const thirdParties = whiteList[siteOwner].resources;
+      thirdPartyWhiteList[counter] = [];
+      for (let i = 0; i < thirdParties.length; i++) {
+        thirdPartyWhiteList[counter].push(thirdParties[i]);
+      }
+      counter++;
+    }
+
+    return {
+      firstPartyWhiteList,
+      thirdPartyWhiteList
+    };
+  },
+
   // Returns true if the request should be stored, otherwise false.
-  shouldStore(tab) {
+  shouldStore(tab, thirdPartyFromRequest) {
     const documentUrl = new URL(tab.url);
     // ignore about:*, moz-extension:* & non-visible tabs (like dev tools)
     if (documentUrl.protocol !== 'about:'
       && documentUrl.protocol !== 'moz-extension:'
       && tab.id !== browser.tabs.TAB_ID_NONE) {
+      // check if third party is on the whitelist (owned by the first party)
+      if (thirdPartyFromRequest) {
+        const firstPartyFromRequest = documentUrl.hostname;
+        for (const firstParty in this.firstPartyWhiteList) {
+          const firstPartyFromList = new RegExp(firstParty);
```

Comment:
Potentially making this change also could to the object check too. Again check and let me know.